### PR TITLE
Add HSTU attention CuTe DSL kernels for Blackwell

### LINKED ISSUE #369: [feature request] Add blackwell HSTU mha kernel.
Meta proposed a new attention kernel for generative recommender [original paper](https://dl.acm.org/doi/10.5555/3692070.3694484).

# HSTU Multi-Head Attention Definition

For each packed sequence and attention head $h$, let
$Q_h, K_h \in \mathbb{R}^{L \times d}$ and
$V_h \in \mathbb{R}^{L \times d_v}$. HSTU attention is defined as:

$$
S_h = \alpha Q_h K_h^{T} + R_h,
$$

$$
A_h = \frac{1}{L_{\mathrm{scale}}}
      \left(M \odot \mathrm{SiLU}(S_h)\right),
$$

$$
O_h = A_h V_h,
$$

$$
\mathrm{HSTU\text{-}MHA}(Q,K,V)
= \mathrm{Concat}(O_1,O_2,\ldots,O_H).
$$

Here:

- $\mathrm{SiLU}(x)=x\,\sigma(x)=\frac{x}{1+e^{-x}}$.
- $\alpha$ is the QK score scale.
- $M_{ij}\in\{0,1\}$ is the attention mask: 1 for a valid QK pair and 0
  otherwise. It can represent causal, local/sliding-window, target-group, or
  arbitrary masks.
- $R_h$ is an optional relative attention bias (RAB).
- $L_{\mathrm{scale}}$ is the sequence normalization factor exposed as
  `scaling_seqlen`; its default public behavior is `max_seqlen_q`.

Unlike standard Transformer attention, HSTU applies SiLU to the QK scores and
does not use row-wise softmax normalization.

We need to implement it with CuTeDSL on Blackwell.

### PR DESCRIPTION
**Pull request by jiayus-nvidia**
## Summary

- Add packed variable-length HSTU attention forward and backward CuTe DSL kernels for Blackwell SM100/SM103 GPUs.
- Support FP16/BF16 head dimensions 64, 128, and 256; full, causal, local, and arbitrary masks; paged-KV forward; and strided or preallocated gradient outputs.
- Add the experimental FE API, lazy top-level exports, documentation, and L0/L1 pytest coverage.
- Apply the repository's existing licensing pattern: NVIDIA-owned implementation, API, and test files use Apache-2.0, while seven retained low-level derived helpers use MIT and preserve their original author notices.
- Make explicit-stream execution allocation- and lifetime-safe, reject cross-device PyTorch streams, and validate output overlap with all read-only metadata.
- Preserve the original HSTU development history while adapting the port to the current `develop` package layout and conventions.

## Validation

- `pre-commit run --files <changed files>`
- `python -m compileall -q python/cudnn/hstu_attention`
- `cd test/python && PYTHONDONTWRITEBYTECODE=1 PYTHONPATH=../../python python -m pytest -p no:cacheprovider -q -m "L0 or L1" fe_api/hstu_attention/`
  - 185 passed on NVIDIA B200 (SM100), CUDA 13.2, cuDNN 9.20, PyTorch 2.11, CUTLASS DSL 4.6.1
- License audit: 19 Apache-2.0 files and 7 MIT files in the HSTU package and tests, with no file-level BSD source in the HSTU tree.

Closes #369

## Summary by CodeRabbit

* **New Features**
  * Added experimental HSTU attention support for Blackwell SM100/SM103 GPUs.
  * Added packed variable-length forward and backward APIs with causal, local, arbitrary, and paged-KV masking.
  * Added SiLU score transformation without softmax, automatic block-sparse metadata, and head-dimension-256 backward support.
  * Exposed HSTU operations through the Python API.

* **Documentation**
  * Added comprehensive API documentation, usage details, limitations, and support information.

* **Tests**
  * Added extensive correctness and integration coverage.

* **Chores**
  * Updated licensing and third-party attribution records.

### CODE REVIEW

**Comment by vedaanta** [comment]
@cudnn-ci-bot run frost,oss

**Review by vedaanta — APPROVED** [review]

**Comment by vedaanta** [comment]
@cudnn-ci-bot run oss
